src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,353 @@
+                preferences are to be stored in.
+            </td>
+            <td>
+                "/add-preferences?view=%view"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <p>
+                    Environment Variable:
+                    <code>GPII_OAUTH2_AUTH_GRANT_TYPE</code>
+                </p>
+
+                <p>
+                    JSON Config Path:
+                    <code>authentication.grant_type</code>
+                </p>
+            </td>
+            <td>
+                The grant type supported by the OAuth2 server.
+            </td>
+            <td>
+                "client_credentials"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <p>
+                    Environment Variable:
+                    <code>GPII_OAUTH2_AUTH_SCOPE</code>
+                </p>
+
+                <p>
+                    JSON Config Path:
+                    <code>authentication.scope</code>
+                </p>
+            </td>
+            <td>
+                The level of permissions that are being requested.
+            </td>
+            <td>
+                "add_preferences"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <p>
+                    Environment Variable:
+                    <code>GPII_OAUTH2_AUTH_CLIENT_ID</code>
+                </p>
+
+                <p>
+                    JSON Config Path:
+                    <code>authentication.client_id</code>
+                </p>
+            </td>
+            <td>
+                The client ID registered with the OAuth2 server.
+            </td>
+            <td>
+                "client_id_first_discovery"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <p>
+                    Environment Variable:
+                    <code>GPII_OAUTH2_AUTH_CLIENT_SECRET</code>
+                </p>
+
+                <p>
+                    JSON Config Path:
+                    <code>authentication.client_secret</code>
+                </p>
+            </td>
+            <td>
+                The client secret registered with the OAuth2 server.
+                Used to securely identify the client with the OAuth2.
+            </td>
+            <td>
+                "client_secret_first_discovery"
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+#### Example: Config File ####
+
+```JSON
+{
+    "securityServer": {
+        "port": "8081",
+        "hostname": "http://10.0.2.2",
+        "paths": {
+            "token": "/access_token",
+            "preferences": "/add-preferences?view=%view"
+        }
+    },
+    "authentication": {
+        "grant_type": "client_credentials",
+        "scope": "add_preferences",
+        "client_id": "client_id_first_discovery",
+        "client_secret": "client_secret_first_discovery"
+    }
+}
+```
+
+#### GPII Preferences Server Specific Configuration ####
```

Comment:
The styling for level 4 headers is non-existant, so this header is virtually invisible.